dask_jobqueue/core.py

```diff
@@ -117,6 +137,12 @@ def __init__(self,
         if env_extra is None:
             env_extra = dask.config.get('jobqueue.%s.env-extra' % self.scheduler_name)
 
+        if cores is not None and threads is not None and processes is not None:
+            raise ValueError("You can only specify two of cores, threads, and processes")
+
+        if total_memory is not None and memory is not None:
+            raise ValueError("You can only specify one of total_memory and memory")
+
```

Comment:
Hm, I think I'm beginning to be incline to remove `threads` and `memory`. At least use `memory` keyword instead of `total_memory` (with a warning comment).